queries/lua/fold.scm

```diff
@@ -0,0 +1,6 @@
+[
+ (for_in_statement)
+ (for_statement)
+ (if_statement)
+ (function_definition)
+ ] @fold
```

Comment:
do we really want to start a new group of query files? We should at least be able to fall back to old behavior. Since it will take ages until all languages are supported and it will take more effort for people who maintain custom languages.

The only problem with the old folding was that one line could have multiple folds. I guess this can still happen when you write everything in one line.

Reply:
Hopefully it should not, but I am diging into the issue rn

Reply:
Should be good now

Reply:
Regarding the new query group thing, I think we should.
With runtime files, vim always had a tednedcy to split things in specialized files, and I think that we should do the same. Plus this allows us to fold, indent, or wathever, on different nodes and allow users to customize however they want

Reply:
We had a feature that was working pretty good without any queries (which was a pretty elegant solution from you). You always have to bear in mind that users have to maintain and write those queries and users starting a new language won't have the time to do that in the first place. It's nice that you can have a better experience with *carefully* tuned configs but we shouldn't force it. A simpler solution is always better.

We will have a feature regression if we don't provide the old fallback. If we force all language maintainers to implement the fold queries in a rush (so not *carefully* tuned). We will have bugs of type: cannot fold for-loops because the maintainer forgot that single node in the query. Would be great if we can provide a nice feature without configuration and perfect one if people are willing to tune it for that specific language. I don't think that people will provide those queries for any language. Right now we have languages with all queries and some with partial support and I guess it'll stay that way for a time.

Reply:
I added a fallback to `locals` scopes, so that is uses that as a default.

Reply:
And in any case, the old solution was buggy, and fixing it would take too much performance, the solution as it is right now, that is `fold.scm` + fallback to `locals.scm` will be good enough.